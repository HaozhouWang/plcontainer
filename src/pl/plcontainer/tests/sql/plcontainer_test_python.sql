select pylog100();
select pylog(10000, 10);
select pybool('f');
select pybool('t');
select pyint(NULL::int2);
select pyint(123::int2);
select pyint(234::int4);
select pyint(345::int8);
select pyfloat(3.1415926535897932384626433832::float4);
select pyfloat(3.1415926535897932384626433832::float8);
select pytext('text');
-- Test that container cannot access filesystem of the host
select pywriteFile();
\! ls -l /tmp/foo
select pyconcat(fname, lname) from users order by 1;
select pyconcatall();
select pynested_call_three('a');
select pynested_call_two('a');
select pynested_call_one('a');
select pyinvalid_function();
select pyinvalid_syntax();
select pylog100_shared();