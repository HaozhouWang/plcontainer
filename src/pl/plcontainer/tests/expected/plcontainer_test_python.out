select pylog100();
 pylog100 
----------
        2
(1 row)

select pylog(10000, 10);
 pylog 
-------
     4
(1 row)

select pybool('f');
 pybool 
--------
 f
(1 row)

select pybool('t');
 pybool 
--------
 t
(1 row)

select pyint(NULL::int2);
ERROR:  exception occurred: (plcontainer.c:478)
DETAIL:  unsupported operand type(s) for +: 'NoneType' and 'int'
select pyint(123::int2);
 pyint 
-------
   124
(1 row)

select pyint(234::int4);
 pyint 
-------
   236
(1 row)

select pyint(345::int8);
 pyint 
-------
   348
(1 row)

select pyfloat(3.1415926535897932384626433832::float4);
 pyfloat 
---------
 4.14159
(1 row)

select pyfloat(3.1415926535897932384626433832::float8);
     pyfloat      
------------------
 5.14159265358979
(1 row)

select pytext('text');
 pytext  
---------
 textbar
(1 row)

-- Test that container cannot access filesystem of the host
select pywriteFile();
       pywritefile        
--------------------------
 wrote foobar to /tmp/foo
(1 row)

\! ls -l /tmp/foo
ls: cannot access /tmp/foo: No such file or directory
select pyconcat(fname, lname) from users order by 1;
 pyconcat  
-----------
 janedoe
 johndoe
 ricksmith
 willemdoe
(4 rows)

select pyconcatall();
      pyconcatall      
-----------------------
 jane,john,rick,willem
(1 row)

select pynested_call_three('a');
 pynested_call_three 
---------------------
 a
(1 row)

select pynested_call_two('a');
      pynested_call_two       
------------------------------
 {'pynested_call_three': 'a'}
(1 row)

select pynested_call_one('a');
                   pynested_call_one                   
-------------------------------------------------------
 {'pynested_call_two': "{'pynested_call_three': 'a'}"}
(1 row)

select pyinvalid_function();
ERROR:  exception occurred: (plcontainer.c:478)
DETAIL:  'module' object has no attribute 'foobar'
select pyinvalid_syntax();
ERROR:  exception occurred: (plcontainer.c:478)
DETAIL:  ('invalid syntax', ('<string>', 7, 1, '\n'))
select pylog100_shared();
 pylog100_shared 
-----------------
               2
(1 row)

